bus/bus.go

```diff
@@ -885,6 +890,103 @@ func (b *Bus) renewContract(ctx context.Context, cs consensus.State, gp api.Goug
 	return newRevision, contractPrice, fundAmount, nil
 }
 
+func (b *Bus) scanHost(ctx context.Context, timeout time.Duration, hostKey types.PublicKey, hostIP string) (rhpv2.HostSettings, rhpv3.HostPriceTable, time.Duration, error) {
```

Comment:
I know everything is there for a reason but this it's still surprising that "simply" scanning a host has to be such a convoluted mess of retries and custom timeouts and all that stuff. I'm also wondering whether we should combine `rhpv2.HostSettings` and `rhpv3.HostPriceTable` because the interface isn't all that clean. We pass a context and a timeout, a hostkey and a host ip and we return data types from both RHPs perhaps soon extending it with a third value.

Reply:
We can revisit that when adding rhp4 scanning. But that's not really related to migrating the endpoint